chore(docs): Add Gatsby Head (#36125)
gatsbyjs/gatsby: docs/docs/reference/built-in-components/gatsby-head.md

---
title: Gatsby Head API
examples:
  - label: Using Gatsby Head
    href: "https://github.com/gatsbyjs/gatsby/tree/master/examples/using-gatsby-head"
---

> Support for the Gatsby Head API was added in `gatsby@4.19.0`.

Gatsby includes a built-in `Head` export that allows you to add elements to the [document head](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/head) of your pages.

> Compared to [react-helmet](https://github.com/nfl/react-helmet) or other similar solutions, Gatsby Head is easier to use, more performant, has a smaller bundle size, and is supporting all latest React features.

Reviewer: ```suggestion
Compared to [react-helmet](https://github.com/nfl/react-helmet) or other similar solutions, Gatsby Head is easier to use, more performant, has a smaller bundle size, and supports the latest React features.
```

## Using Gatsby Head in your page

By exporting a named function called `Head` you can set the metadata for a page:

```jsx:title=src/pages/index.jsx
import * as React from "react"

const Page = () => <div>Hello World</div>
export default Page

export function Head() {
  return (
    <title>Hello World</title>
  )
}
```

You can also use the arrow function syntax:

```jsx
export const Head = () => <title>Hello World</title>
```

When defining multiple metatags use React Fragments:

```jsx
export const Head = () => (
  <>
    <title>Hello World</title>
    <meta name="description" content="Hello World" />
  </>
)
```

### Usage notes

You'll need to be aware of these things when using Gatsby Head:

- You can only define the `Head` export inside a page, not in a component.
- The contents of Gatsby Head get cleared upon unmounting the page, so make sure that each page defines what it needs in its `<head>`.
- The `Head` function needs to return valid JSX.
- Valid tags inside the `Head` function are: `link`, `meta`, `style`, `title`, `base`, and `noscript`.

> - If you want to add `<script />` tags to your pages, use the [Gatsby Script Component](/docs/reference/built-in-components/gatsby-script/).

Reviewer: ```suggestion
- For most `<script>` tag use cases you should use the [Gatsby Script Component](/docs/reference/built-in-components/gatsby-script/) in your pages or components.
```

> - For most `<script>` tag use cases you should use the [Gatsby Script Component](/docs/reference/built-in-components/gatsby-script/) in your pages or components.

Reviewer: ```suggestion
- Data block `<script>` tags such as `<script type="application/ld+json">` can go in the `Head` function, but dynamic scripts are better loaded with the [Gatsby Script Component](/docs/reference/built-in-components/gatsby-script/) in your pages or components.
```
